pkg/crypto/tinkcrypto/primitive/composite/ecdh1pu/ecdh1pu_aes_aead_public_key_manager.go

```diff
@@ -0,0 +1,140 @@
+func (km *ecdh1puPublicKeyManager) Primitive(serializedKey []byte) (interface{}, error) {
+	if len(serializedKey) == 0 {
+		return nil, errInvalidECDH1PUAESPublicKey
+	}
+
+	ecdh1puPubKey := new(ecdh1pupb.Ecdh1PuAeadPublicKey)
+
+	err := proto.Unmarshal(serializedKey, ecdh1puPubKey)
+	if err != nil {
+		return nil, errInvalidECDH1PUAESPublicKey
+	}
+
+	_, err = km.validateKey(ecdh1puPubKey)
+	if err != nil {
+		return nil, errInvalidECDH1PUAESPublicKey
+	}
+
+	var recipientsKeys []*composite.PublicKey
+
+	for _, recKey := range ecdh1puPubKey.Params.KwParams.Recipients {
+		e := km.validateRecKey(recKey)
+		if e != nil {
+			return nil, errInvalidECDH1PUAESPublicKey
+		}
+
+		pub := &composite.PublicKey{
+			KID:   recKey.KID,
+			Type:  recKey.KeyType.String(),
+			Curve: recKey.CurveType.String(),
+			X:     recKey.X,
+			Y:     recKey.Y,
+		}
+
+		recipientsKeys = append(recipientsKeys, pub)
+	}
+
+	rEnc, err := composite.NewRegisterCompositeAEADEncHelper(ecdh1puPubKey.Params.EncParams.AeadEnc)
+	if err != nil {
+		return nil, err
+	}
+
+	ptFormat := ecdh1puPubKey.Params.EcPointFormat.String()
+
+	return subtle.NewECDH1PUAEADCompositeEncrypt(recipientsKeys, ptFormat, rEnc, commonpb.KeyType_EC), nil
+}
+
+// DoesSupport indicates if this key manager supports the given key type.
+func (km *ecdh1puPublicKeyManager) DoesSupport(typeURL string) bool {
+	return typeURL == ecdh1puAESPublicKeyTypeURL
+}
+
+// TypeURL returns the key type of keys managed by this key manager.
+func (km *ecdh1puPublicKeyManager) TypeURL() string {
+	return ecdh1puAESPublicKeyTypeURL
+}
+
+// NewKey is not implemented for public key manager.
+func (km *ecdh1puPublicKeyManager) NewKey(serializedKeyFormat []byte) (proto.Message, error) {
+	return nil, fmt.Errorf("ecdh1pu_public_key_manager: NewKey not implemented")
+}
+
+// NewKeyData is not implemented for public key manager.
+func (km *ecdh1puPublicKeyManager) NewKeyData(serializedKeyFormat []byte) (*tinkpb.KeyData, error) {
+	return nil, fmt.Errorf("ecdh1pu_public_key_manager: NewKeyData not implemented")
+}
+
+// validateKey validates the given ECDHESPublicKey.
+func (km *ecdh1puPublicKeyManager) validateKey(key *ecdh1pupb.Ecdh1PuAeadPublicKey) (elliptic.Curve, error) {
+	err := keyset.ValidateKeyVersion(key.Version, ecdh1puAESPublicKeyVersion)
+	if err != nil {
+		return nil, fmt.Errorf("ecdh1pu_publie_key_manager: invalid key: %s", err)
+	}
+
+	return validateKeyFormat(key.Params)
+}
+
+// validateRecKey validates the given recipient's ECDHESPublicKey.
+func (km *ecdh1puPublicKeyManager) validateRecKey(key *ecdh1pupb.Ecdh1PuAeadRecipientPublicKey) error {
+	err := keyset.ValidateKeyVersion(key.Version, ecdh1puAESPublicKeyVersion)
+	if err != nil {
+		return fmt.Errorf("ecdh1pu_public_key_manager: invalid key: %s", err)
+	}
+
+	_, err = composite.GetKeyType(key.KeyType.String())
+	if err != nil {
+		return err
+	}
+
+	_, err = hybrid.GetCurve(key.CurveType.String())
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
```

Comment:
can be simplified 

```
_, err = hybrid.GetCurve(key.CurveType.String())

return err
```

Reply:
if we do this.. then the error message is not going to be wrapped, I will add wrapping message instead

Reply:
the last return is not necessary to wrap

Reply:
to be consistent with the all error messages in this file, we need to wrap it to start with `ecdh1pu_public_key_manager:`